datafusion/expr/src/udf.rs

```diff
@@ -293,14 +293,17 @@ where
 
 /// Arguments passed to [`ScalarUDFImpl::invoke_with_args`] when invoking a
 /// scalar function.
-pub struct ScalarFunctionArgs<'a> {
+pub struct ScalarFunctionArgs<'a, 'b> {
     /// The evaluated arguments to the function
     pub args: Vec<ColumnarValue>,
+    /// Field associated with each arg, if it exists
```

Comment:
very minor but i found it strange that `a` was used second (as in I would expect 
```rust
    pub arg_fields: Vec<Option<&'a Field>>,
...
    pub return_field: &'b Field,
```